nilearn/decomposition/_base.py

```diff
@@ -32,6 +33,32 @@
 from nilearn.surface import SurfaceImage
 
 
+def _warn_ignored_surface_masker_params(estimator):
+    """Warn about parameters that are ignored by SurfaceMasker.
```

Comment:
note that something similar should probably be generalized for other functions/methods that accept both volume and surface data, but use some parameters for volume only

Reply:
Yeah I think we were looking at this and this function seemed like the most appropriate place for this: 

https://github.com/nilearn/nilearn/blob/ea4cd19f777ba8bd84018904154b092ff5cd3252/nilearn/_utils/masker_validation.py#L15

But we decided to keep that for another PR